### ISSUE TITLE
Merlin 2.5.0 for Emacs informs "merlin: no result" when querying type information

### ISSUE THREAD

**Issue by taquangtrung** [the-lambda-church/merlin#549]
I encountered the above problem when using Merlin 2.5.0 for Emacs with both OCaml 4.02.3 and 4.03.0:
Merlin always report "merlin: no result" when I try to query type of the expression by either using `C-c C-t` or invoking `merlin-type-enclosing`

Can you help to check if this is a problem of Merlin 2.5.0?

I use Emacs Prelude (https://github.com/bbatsov/prelude) and install both Ocaml, Merlin by using Opam.
This problem doesn't happen in the older version Merlin 2.3.1.

Thanks!

**Comment by let-def** [comment]
This is likely because you use the merlin-mode from melpa rather than the one provided with merlin distribution.

merlin-mode from melpa should be synchronized soon, but it is not the preferred way of distribution.

**Comment by taquangtrung** [comment]
Thanks for your update. I'm looking forward to see the synchronized merlin :-D

**Comment by taquangtrung** [comment]
I've just tried to directly install Merlin from Github but still encountered the same problem. Can you please double check? I also attached to pictures as follows:

In the this picture, Merlin fails to guess the type of `bar` expression.

![merlin-1](https://cloud.githubusercontent.com/assets/193967/16566400/a1f42eb6-4246-11e6-98ef-4ee6652ef76a.png)

I use the latest Gerlin with git-id: `b2926cf6235827c0242797775188b8f0420386c5`

![merlin-2](https://cloud.githubusercontent.com/assets/193967/16566401/a6adcebc-4246-11e6-9897-84a5bc63c064.png)

FYI, I used Emacs 24.5.1, complied from source code.

**Comment by let-def** [comment]
What's the result of `M-x find-library merlin`?

**Comment by taquangtrung** [comment]
`M-x find-library merlin` return a version of Merlin from Dec 2015.
I removed that library from elpa and it works perfectly with the newest Merlin from Github.
Thank you very much! :+1:

**Comment by awonnacott** [comment]
I was having this issue as well, and the solution on this page was not helping. I ran "show type of an expression" and got the following error:

    (merlin) backtrace: 
    Merlin failed with error: "Error: /usr/lib/ocaml/pervasives.cmi
    is not a compiled interface for this version of OCaml.
    It seems to be for a newer version of OCaml."

This was caused by out-of-date opam packages - it was fixed by running `opam update` then `opam upgrade`, then restarting merlin in emacs.

I add this here because "merlin: no result" was all that was shown when querying type around the cursor, and this page is the top Google result for that issue.
